docs/technical/mock-classifier-technical-architecture.md

```diff
@@ -58,50 +60,63 @@ These resolutions will be hard coded into the mock classifier and will not requi
 
 Here are some examples:
 
-- An input text of `Please return the {{social}} minsitry` will return `social`
-- An input text of `{{environment}}` will return `environment`
-- An input text of `{{rural}} please` will return `rural`
+- An input text of `Please return the {{social}} minsitry` will call back with `social`
+- An input text of `{{environment}}` will call back with `environment`
+- An input text of `{{rural}} please` will call back with `rural`
+
+Each token detected will result in a separate call back to DMR. For example, input text of `I want to see {{rural}}{{social}} and {{environment}}` will return `rural`, `social` and `environment` as three separate DMR call backs.
 
-Spaces are not permitted so an input text of `Please return {{ environment}}` would return an error code.
+We want the Classifier to support random ministry selection to help with testing, therefore is the `{{random}}` token is used, the Classifier will randomly select and return one of the 11 ministries. For example, an input text of `Give me a {{random}} one` will return a randomly selected ministry.
 
-We want the Classifier to support random ministry selection to help with testing, therefore is the `{{random}}` token is used, the Classifier will randomly select and return one of the 11 ministries.
+Spaces are not permitted so an input text of `Please return {{ environment}}` would be treated as an error case.
 
-For example, an input text of `Give me a {{random}} one` will return a randomly selected ministry.
+If a token is detected with unrecognised text (i.e. text that does not exactly map to one of the labels defined in the Ministries section or the word `random`), this will be treated as an error case. For example, the input text of `Please return {{educationandresearch}}` would be an error as would `{{Please return {{education,social}}`.
 
 ## API Design
 
 The Classifier will be a REST api with just a single endpoint which is `/classifications`.
 
 #### Request
 
-- Accepts only `POST` requests. Other http verbs will result in `404/NotFound` responses.
-- Expects an `input` parameter which contains the text representing the input text. This will be defined in a JSON request body, for example
+Accepts only `POST` requests. Other http verbs will result in `404/NotFound` responses.
+
+Expects an `message` parameter which is an array of strings. The strings contains the text representing the input text. This will be defined in a JSON request body, for example
 
 ```json
 {
-"input":"I want to register my child at school"
+"message":["I want to register my child at school"]
 }
 ```
 
 or
 
 ```json
 {
-"input":"Please return the {{social}} minsitry"
+"message":["Please return the {{social}} minsitry"]
 }
 ```
 
+Other parameters may also be required in order for the Classifier to call back to DMR when the processing is completed (call-back uri for example). Please consult the DMR design documentation for details of these parameters.
```

Comment:
Can I suggest that we spell "Callback" as a single word term really to stay consistent with industry standards.

```suggestion
Other parameters may also be required in order for the Classifier to callback to DMR when the processing is completed (callback URI for example). Please consult the DMR design documentation for details of these parameters.
```